legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -167,19 +167,24 @@ def search_businesses():
         draft_query = db.session.query(Filing).filter(
             and_(Filing.temp_reg.in_(identifiers), Filing.business_id.is_(None)))
 
-        # parse results
         bus_results = [x.json(slim=True) for x in bus_query.all()]
-        draft_results = [
-            {
-                'identifier': x.temp_reg,
-                'legalType': x.json_legal_type,
-                **({'nrNumber': x.json_nr} if x.json_nr else {}),
-                'legalName': (x.filing_json.get('filing', {})
-                              .get(x.filing_type, {})
-                              .get('nameRequest', {})
-                              .get('legalName')),
-                'draftType': Filing.FILINGS.get(x.filing_type, {}).get('temporaryCorpTypeCode')
-            } for x in draft_query.all()]
+
+        draft_results = []
+        for draft_dao in draft_query.all():
+            draft = {
+                'identifier': draft_dao.temp_reg,  
+                'legalType': draft_dao.json_legal_type
+            }
+            if draft_dao.json_nr:
+                draft['nrNumber'] = draft_dao.json_nr
+            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
+                                        .get(draft_dao.filing_type, {})
+                                        .get('nameRequest', {})
+                                        .get('legalName'))
+            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
+            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
```

Comment:
draft['legalName'] can be None for Incorporation Application (if its numbered)

Reply:
will fix

Reply:
Can IA have draftType = ATMP?

Reply:
I see, we could put in for IA as well.. hmm one sec

Reply:
I think we'd have to put most of this code in here:

https://github.com/bcgov/bcrs-shared-components/blob/main/src/modules/corp-type-module/corp-type-module.ts
To compute IA's numbered text
```
            if draft['draftType'] == 'TMP' and not draft_dao.json_nr and draft_dao.json_legal_type in ('BC', 'BEN','ULC', 'CCC'):
                draft['legalName'] = generate_numbered_ia_name()
```

Reply:
``` 
draft_results = []
        ia_numbered_dict = {
            'BC': 'Numbered Limited Company',
            'BEN': 'Numbered Benefit Company',
            'ULC': 'Numbered Unlimited Liability Company',
            'CCC': 'Numbered Community Contribution Company'
        }
        for draft_dao in draft_query.all():
            draft = {
                'identifier': draft_dao.temp_reg,  
                'legalType': draft_dao.json_legal_type
            }
            if draft_dao.json_nr:
                draft['nrNumber'] = draft_dao.json_nr
            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
                                        .get(draft_dao.filing_type, {})
                                        .get('nameRequest', {})
                                        .get('legalName'))
            draft['draftType'] = Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode')
            if draft['draftType'] == 'ATMP' and draft['legalName'] is None:
                draft['legalName'] = 'Numbered Amalgamated Company'
            if draft['draftType'] == 'TMP' and not draft_dao.json_nr:
                draft['legalName'] = ia_numbered_dict.get(draft_dao.json_legal_type, None)
            draft_results.append(draft)
```

Reply:
something like that?

Reply:
```
    BUSINESSES = {
        LegalTypes.BCOMP: {
            'numberedLegalNameSuffix': 'B.C. LTD.',
            'numberedDescription': 'Numbered Benefit Company'
        },
        LegalTypes.COMP: {
            'numberedLegalNameSuffix': 'B.C. LTD.',
            'numberedDescription': 'Numbered Limited Company'
        },
        LegalTypes.BC_ULC_COMPANY: {
            'numberedLegalNameSuffix': 'B.C. UNLIMITED LIABILITY COMPANY',
            'numberedDescription': 'Numbered Unlimited Liability Company'
        },
        LegalTypes.BC_CCC: {
            'numberedLegalNameSuffix': 'B.C. COMMUNITY CONTRIBUTION COMPANY LTD.',
            'numberedDescription': 'Numbered Community Contribution Company'
        }
    }
    ```